src/rgw/rgw_sal_motr.h

```diff
@@ -40,6 +44,215 @@ class MotrStore;
 #define RGW_MOTR_BUCKET_HD_IDX_NAME   "motr.rgw.bucket.headers"
 //#define RGW_MOTR_BUCKET_ACL_IDX_NAME  "motr.rgw.bucket.acls"
 
+// Notes for Motr FDMI:
+// (1) Currently FDMI can only trigger events for index operation, not object
+//     operation.
+// (2) When an index op event is generated, the FDMI filter picks up the events
+//     matching the pre-defined rules. The FDMI event contains the key/value of
+//     the index operation. Currently, fdmi can only do sub string matching on
+//     the value part, not key part, and can't specify which index to watch,
+//     the filter rule is applied to all indices.
+//     this 
+// (3) The FDMI event is delivered to an FDMI application which defines a
+//     callback function to process FDMI events.
+//
+// Implementation of object watch-notify using Motr FDMI. 
+// (1) An FDMI filter is added (in Motr configuration yaml file) specifying
+//     a matching sub-string (marker).
+// (2) A notifier creates a set of indices which are used to trigger FDMI
+//     events and to pass messages.
```

Comment:
This is a bit confusing: who is that notifier in our case that creates the set of indexes?

Reply:
In metadata cache's case, the cache has a notifier which attaches notification message to the value part of an PUT operation to the metadata index. And the cache doesn't create the set of indices, it uses the metadata index to trigger FDMI events and pass on notification messages.

The notifier class tries to generalise the usage of sending notification (in FDMI's implementation, trigger FDMI events and pass on notification in FDMI events). So it includes a set of dedicated indices for passing notification. If there is other indices we can piggyback the notification like caching, those indices won't be used.